src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/message/TextComponent.kt

```diff
@@ -228,4 +231,75 @@ class TextComponent {
                 )
             )
     }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    private fun stringToComponent(string: String): MCBaseTextComponent {
+        val buffer = StringBuilder()
+        val comp = MCBaseTextComponent("")
+        var style = ChatStyle()
+        var i = 0
+        while (i < string.length) {
+            if (i < string.length - 1 && formatRegex.matches(string.substring(i..i + 1))) {
+                // add the previous component as a sibling of the base component
+                val prevText = MCBaseTextComponent(buffer.toString()).apply { chatStyle = style.createDeepCopy() }
+                buffer.clear()
+                comp.appendSibling(prevText)
+
+                // Update the style local var
+                i++
+                when {
+                    string[i] in '0'..'f' -> style.color = EnumChatFormatting.values()[string[i].digitToInt(16)]
+                    string[i] == 'k' -> style.obfuscated = true
+                    string[i] == 'l' -> style.bold = true
+                    string[i] == 'm' -> style.strikethrough = true
+                    string[i] == 'n' -> style.underlined = true
+                    string[i] == 'o' -> style.italic = true
+                    string[i] == 'r' -> style = ChatStyle()
+                }
+            } else if (URL_REGEX.matchesAt(string, i)) {
+                // add the previous component as a sibling of the base component
+                val prevText = MCBaseTextComponent(buffer.toString()).apply { chatStyle = style.createDeepCopy() }
+                buffer.clear()
+                comp.appendSibling(prevText)
+
+                val link = URL_REGEX.matchAt(string, i)!!.value
+                i += link.length - 1
```

Comment:
Why the `- 1` here? Maybe I'm dumb but that doesn't seem necessary, if the link is 10 characters long, we'd want to start searching 10 characters after the start.

Reply:
Ahh right that makes sense